src/c++/perf_analyzer/infer_context.h

```diff
@@ -48,6 +49,11 @@ struct ThreadStat {
   // TODO REFACTOR TMA-1046 -- This should be in the InferContext class
   // The statistics of the InferContext
   std::vector<cb::InferStat> contexts_stat_;
+  std::chrono::_V2::steady_clock::time_point start_time;
+  // The total amount of time spent sleeping or waiting
+
+  IdleTimer idle_timer;
```

Comment:
Why keep this in ThreadStat? Why not in LoadWorker? Worker already = 1 thread.

Reply:
Also unrelated but ThreadStat doesn't seem like it should go in InferContext. One worker can have multiple InferContexts, but One worker only has one ThreadStat.

Reply:
Yes, Worker == 1 thread, and each worker has one ThreadStat. It is the stats for the thread, which is why I am putting this new stat for the thread in it.

InferContexts do not own ThreadStat. They are passed one. I do agree that there is still cleanup work to be done here. TMA-1046 is one of those stories, and is somewhat related to your comment.

Reply:
I do need to think a bit more to make sure the contexts can't screw with each other's usage of the timer though. I'm thinking I need to move [this close bracket](https://github.com/triton-inference-server/client/pull/249/files#diff-6e28646eb27305f306e6805023d5013a4e5a0c167367a67fd58811f855e3aa40R121) down to line 134 so that [this mutex](https://github.com/triton-inference-server/client/pull/249/files#diff-6e28646eb27305f306e6805023d5013a4e5a0c167367a67fd58811f855e3aa40R112) applies to the infer request as well. Do you agree?